### PR TITLE
Added Texture.DEFAULT_ANISOTROPY

### LINKED ISSUE #25010: Ability to control `texture.anisotropy` from `GLTFLoader`
**Is your feature request related to a problem? Please describe.**

We were often wondering why textures in three.js looked more blurry than in other engines. See for example the files in https://github.com/google/model-viewer/issues/1505. Turns out that while generally texture anisotropy can be set, GLTFLoader currently doesn't expose a way to control it, so glTF viewers generally look "kind of blurry" from steep angles.  
 
I'm happy to work on a PR for this but would like to align on possible solutions first.  

The renderer has `renderer.capabilities.getMaxAnisotropy` which contains the maximum allowed anisotropy, but I think being able to pass in an override for this into the loader would be nice instead of just setting the max (and there's currently no access to the renderer as far as I can see, minus spawning a new one).  

Currently I'm just adding `texture.anisotropy = 4;` in GLTFLoader.loadTextureImage so that it applies to textures loaded via extensions as well.

**Describe the solution you'd like**

Some way to override texture anisotropy as used by GLTFLoader.  

**Describe alternatives you've considered**

- Adding some texture loading extension. Complexity would be that this would still need to work with all other texture loading extensions, so not entirely sure how that would work.  

- Iterating over all materials and all textures in the glTF file and setting anisotropy there. Harder than it should be in my opinion but maybe I'm missing a convenient way to do that.

If there's a good way to set this without changing GLTFLoader I'm all ears :)

### PR DESCRIPTION
**Pull request by WestLangley**
Fixed #25010

Perhaps this is an acceptable solution...

### CODE REVIEW

**Comment by WestLangley** [comment]
Oh wait. I see @takahirox just suggested the same idea. :-)

File: src/textures/Texture.js [hunk 1]
```diff
@@ -304,5 +304,6 @@ class Texture extends EventDispatcher {
 
 Texture.DEFAULT_IMAGE = null;
 Texture.DEFAULT_MAPPING = UVMapping;
+Texture.DEFAULT_ANISOTROPY = 1;
```
Comment (LeviPesin) [inline src/textures/Texture.js:307]: Hm, why the case differs between `Object3D.DefaultUp`/`Object3D.DefaultMatrixAutoUpdate` and `Texture.DEFAULT_IMAGE`/`Texture.DEFAULT_MAPPING`/`Texture.DEFAULT_ANISOTROPY`?

**Comment by Mugen87** [comment]
I've also thought about adding this change when reading https://github.com/mrdoob/three.js/issues/25010 but I'm afraid this opens the door for adding even more global defaults. Devs could argue to add the same for other default values and it's hard to argue against this if we merge this PR.

To me, I think the presented workaround in https://github.com/mrdoob/three.js/issues/25010#issuecomment-1327017463 sounds sufficient. An additional traverse should not be noticeable in terms of loading time.

**Comment by takahirox** [comment]
Another solution may be...

* Test `.anisotropy > 1` and check visual quality and performance
* If visual quality benefit is big and performance penalty is small, change the hardcoded default `.anisotropy` to for example 4.

**Comment by hybridherbst** [comment]
Regarding "opening tthe door for more globals", out of curiosity, I think it would still be on a case-by-case basis, right? And in some cases it may well make sense. Anisotropy as an example of a typical per-app default is kind of a good example for something that could be a global.

In Unity Anisotropy is also one of the few quality globals:
image
(it has options "Disabled", "Per Texture", "Forced On")

**Review by Mugen87 — APPROVED** [review]
Okay, I'm convinced. The addition indeed make sense in this case.

File: src/textures/Texture.js (hunk 1 above)
Comment (Mugen87) [reply src/textures/Texture.js:307]: I'm not sure but I think @mrdoob lately stated his preference for this nomenclature. Unfortunately, I don't find the discussion right now.

For the time being, it's sufficient when we adhere to naming conventions per file. At a later point, we can unify the syntax.

Comment (mrdoob) [reply src/textures/Texture.js:307]: It probably should be `Object3D.DEFAULT_UP` and `Object3D.DEFAULT_MATRIXAUTOUPDATE`...

@LeviPesin Are these the only ones that would need to be changed?

Comment (WestLangley) [reply src/textures/Texture.js:307]: I can make these changes for consistency if @mrdoob approves.
```js
// from
Object3D.DefaultUp = /*@__PURE__*/ new Vector3( 0, 1, 0 );
Object3D.DefaultMatrixAutoUpdate = true;
Object3D.DefaultMatrixWorldAutoUpdate = true;

// to
Object3D.DEFAULT_UP = /*@__PURE__*/ new Vector3( 0, 1, 0 );
Object3D.DEFAULT_MATRIXAUTOUPDATE = true;
Object3D.DEFAULT_MATRIXWORLDAUTOUPDATE = true;

// from
Euler.DefaultOrder = 'XYZ';
Euler.RotationOrders = [ 'XYZ', 'YZX', 'ZXY', 'XZY', 'YXZ', 'ZYX' ];

// to
Euler.DEFAULT_ORDER = 'XYZ';
Euler.RotationOrders // wait. why is this here? leave as-is -- or deprecate it?

// as-is
Texture.DEFAULT_IMAGE = null;
Texture.DEFAULT_MAPPING = UVMapping;
Texture.DEFAULT_ANISOTROPY = 1;
```

Comment (Mugen87) [reply src/textures/Texture.js:307]: > Euler.RotationOrders // wait. why is this here? leave as-is -- or deprecate it?

I vote to remove it. Possible euler rotation orders should be clear without defining them in an array.

Comment (WestLangley) [reply src/textures/Texture.js:307]: Completely remove it with an added migration note -- or add a 10-month deprecation warning and retain it?

**Pull request merged**